core/utils.ts

```diff
@@ -189,6 +186,10 @@ export function setNameAndTarget(
   overrideDatabase?: string
 ) {
   action.target = target(session.config, name, overrideSchema, overrideDatabase);
+  if (action.config) {
+    action.config.target = target(session.canonicalConfig, name, overrideSchema, overrideDatabase);
+    return;
+  }
```

Comment:
i'm confused. why do we need this / what's it for / should it really be here, as opposed to the callsite?